### ISSUE TITLE
pytest - analyse test coverage

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#48]
Sometimes, pytest checker gets called with eg "--cov-report=term-missing\"

It would be good if the LLM could get the test coverage

First analysis what could be done:

---
---

Got it — you’re doing "vibe coding" (explore & hack, then let an LLM help clean up), and you want the **test coverage report** to flow back into the LLM in a useful way.

Here are the practical options:

---

### 1. Use `pytest-cov` to generate structured output

```bash
pytest --cov=your_package --cov-report=term-missing --cov-report=json:coverage.json
```

This gives you:

* **human-readable coverage** in the terminal
* **machine-readable JSON** (`coverage.json`), which you can feed back to the LLM.

JSON is best because it avoids hallucinations when parsing plain text.

---

### 2. Normalize before sending to the LLM

LLMs don’t need the whole raw JSON — just the essentials. A preprocessing script can extract:

* File paths
* Coverage percentage
* Missing lines (with context)

Example Python snippet:

```python
import json

with open("coverage.json") as f:
    data = json.load(f)

summary = []
for file, info in data["files"].items():
    summary.append({
        "file": file,
        "covered": info["summary"]["covered_lines"],
        "missing": info["summary"]["missing_lines"],
        "percent": info["summary"]["percent_covered"],
        "missing_lines": info["missing_lines"]
    })

print(summary)
```

You can then feed that compact structure to the LLM.

---

### 3. Provide context with source snippets

Coverage is most useful when the LLM can *see* the missing lines in context.
So instead of just sending "missing lines: \[42, 43, 88]", grab the actual source lines (say ±5 lines around each missing one). That way the LLM can propose tests directly.

---

### 4. Workflow

* Run `pytest --cov` with JSON output.
* Preprocess into a compact “coverage digest” with missing lines + snippets.
* Send that digest to the LLM.
* Ask the LLM: *“Write tests for the uncovered cases.”*

---

### 5. Alternative formats

* `pytest --cov-report=xml:coverage.xml` → machine-readable XML, but JSON is usually easier.
* Tools like [`coverage.py` API](https://coverage.readthedocs.io/en/latest/api.html) let you integrate coverage directly in Python, which is great if you want to automate LLM prompting inside your dev loop.

---

⚡ Recommended: JSON + preprocessing → LLM. That way you keep it clean, structured, and minimal.

---

Do you want me to sketch a **ready-to-use Python script** that takes the `coverage.json`, extracts uncovered lines with context, and prints an LLM-friendly summary?
